File: eureka/android/library/src/main/java/com/lifeplus/Util/LpLogger.java
package com.lifeplus.Util;

import android.util.Log;

import com.lifeplus.EventEmittersToReact;
import com.lifeplus.Pojo.LoggerStruct;

import java.util.Timer;
import java.util.TimerTask;

public class LpLogger {
    private static long _logSrlNo = 0;

    public static long getNewSrlNo() {
        return _logSrlNo++;
    }

    public static void logInfo(final LoggerStruct mLog) {
        TimerTask timerTask = new TimerTask() {
            public void run() {
                String request = mLog.getJSONStr();
                EventEmittersToReact.getInstance().debugLog(request);
            }
        };
        Log.i("logInfo", mLog.getJSONStr());
        Timer mLogTimer = new Timer();
        mLogTimer.schedule(timerTask, 100);
    }

    public static void logError(final LoggerStruct mLog) {
        TimerTask timerTask = new TimerTask() {
            public void run() {
                String request = mLog.getJSONStr();
                EventEmittersToReact.getInstance().debugLog(request);
            }
        };
        Log.i("logError", mLog.getJSONStr());
        Timer mLogTimer = new Timer();
        mLogTimer.schedule(timerTask, 100);
    }
}

File: eureka/ios/nativeLib/nativeLib/Utils/LpLogger.swift
//
//  LpLogger.swift
//  eureka
//
//  Created by work on 04/02/21.
//

import Foundation

public class LpLogger {
    private static var _logSrlNo: Int64 = 0
    internal static var logProvider: (LoggerStruct) -> Void = { logerStruct in
        EventEmittersToReact.debugLog(logerStruct)
    }
  
    internal static func getNewSrlNo() -> Int64 {
        _logSrlNo += 1
        return _logSrlNo
    }

    public static func logInfo(_ pLog:LoggerStruct) {
        if (Global.isDebugMode()) {
            logProvider(pLog)
        }
    }

    internal static func logError(_ pLog: LoggerStruct) {
        logProvider(pLog)
    }
}
